presto-main/src/main/java/com/facebook/presto/sql/planner/CompilerConfig.java

```diff
@@ -37,4 +40,30 @@ public CompilerConfig setExpressionCacheSize(int expressionCacheSize)
         this.expressionCacheSize = expressionCacheSize;
         return this;
     }
+
+    public int getScanNodeLimit()
+    {
+        return this.scanNodeLimit;
+    }
+
+    @Config("planner.max-scan-node-in-plan")
+    @ConfigDescription("Maximum number of scan nodes in logical plan, throw exception when exceed if plan-node-limit-enabled is set true")
+    public CompilerConfig setScanNodeLimit(int num)
+    {
+        this.scanNodeLimit = num;
+        return this;
+    }
+
+    public boolean getPlanNodeLimitEnabled()
+    {
+        return this.planNodeLimitEnabled;
+    }
+
+    @Config("planner.plan-node-limit-enabled")
```

Comment:
call this `planner.scan-node-limit-enabled` (and update corresponding variables and method names)

Reply:
Changed to `leaf-node-limit-enabled`.